docs: add a handbook overview page (#18)
cusp-ai-oss/kUPS: docs/handbook.md

# Handbook

> A tour of the primitives *k*UPS is built from. Each chapter covers one primitive: what it is and what forced it into the design.

Reviewer: ```suggestion
A tour of the primitives *k*UPS is built from. Each chapter covers one primitive: what it is and why it's included in the design.
```
Just softening the language a little, not critical.
